### DISCUSSION TITLE
How to fix phantom "Failed to estimate balance changes. This transaction may be unsafe"

### DISCUSSION THREAD

**Discussion by shirecoding** [phantom/docs#209] [Transaction Simulation]
### dApp Name

Community Crossover

### dApp URL

running locally on development

### Program ID

running locally on development

### Steps to reproduce

I have an issue where I create a transaction on the backend and pay for the transaction fees, then I serialize and send the transaction to the frontend for the user to sign. It works however I get this Prompt from phantom always and it takes a long time after clicking confirm - but the transaction goes through.

How do you properly make a transaction such that it is safe and able to show the balance changes? should be 0 here since the transaction/rent fees are already paid for by a backend wallet.

Thanks!

Here is the transaction:

```
{
    "recentBlockhash": "BqTPpSb74yPSikcVy1p3e5BmyGpDqY8cbmN7g4D2pUvd",
    "feePayer": "L8WnqU2xmTxyQdeNdMDJonBEueBcMyfH33zJgsx9xz9",
    "nonceInfo": null,
    "instructions": [
        {
            "keys": [
                {
                    "pubkey": "4oLzhJMmhiiVWi6H42WmkeTaHn6kfAMEezY5BCZpeFCu",
                    "isSigner": false,
                    "isWritable": true
                },
                {
                    "pubkey": "B5nXXypvm4xeGtAUAPepYLfJV5PsfKvZj4c7SGD3ri53",
                    "isSigner": false,
                    "isWritable": true
                },
                {
                    "pubkey": "2za7tzXbWeJiRhCiaWbdZbxUKpawjWgNpzksptiSguZF",
                    "isSigner": false,
                    "isWritable": true
                },
                {
                    "pubkey": "E5CdwMifQH9CeHGjNf8YefDb6EDqJK84zuVNqGHks8tR",
                    "isSigner": false,
                    "isWritable": false
                },
                {
                    "pubkey": "6xZ4UjShw6qfoR15BxyMfbukWcHN4nVFvCNDQNscggXM",
                    "isSigner": false,
                    "isWritable": true
                },
                {
                    "pubkey": "FgwM9n3damZc4CG8VZU9tTrJ7nWLA6T51URNZ2fnh7Gm",
                    "isSigner": false,
                    "isWritable": false
                },
                {
                    "pubkey": "L8WnqU2xmTxyQdeNdMDJonBEueBcMyfH33zJgsx9xz9",
                    "isSigner": true,
                    "isWritable": true
                },
                {
                    "pubkey": "a7PQyMNYxikUT5jAxeLhSz5c1PWMZEDqXBW9HbSnE4x",
                    "isSigner": true,
                    "isWritable": false
                },
                {
                    "pubkey": "ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL",
                    "isSigner": false,
                    "isWritable": false
                },
                {
                    "pubkey": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                    "isSigner": false,
                    "isWritable": false
                },
                {
                    "pubkey": "11111111111111111111111111111111",
                    "isSigner": false,
                    "isWritable": false
                }
            ],
            "programId": "B2ejsK7m3eYPerru92hS73Gx7sQ7J83DKoLHGwn6pg5v",
            "data": [
                42,
                192,
                142,
                50,
                73,
                168,
                119,
                96,
                1,
                0,
                0,
                0,
                0,
                0,
                0,
                0
            ]
        }
    ],
    "signers": [
        "L8WnqU2xmTxyQdeNdMDJonBEueBcMyfH33zJgsx9xz9",
        "a7PQyMNYxikUT5jAxeLhSz5c1PWMZEDqXBW9HbSnE4x"
    ]
}
```

Why does it show a network fee when i've already paid for the user's transaction in the backend? Also how do you properly implement this with the warning "Failed to simulate the results of this request. Confirming is unsafe and may lead to losses.". I think no one will be using dapps with this warning.

<img width="350" alt="Screenshot 2024-04-04 at 11 32 32 AM" src="https://github.com/phantom/docs/assets/69025414/ba3883f3-78e1-4850-9864-e9e80ed67169">

Thanks!

**Comment by adamdelphantom** [comment]
Hi @shirecoding,

Thank you for writing in!

If the user has to sign and broadcast a transaction there's no way that I'm aware of to bypass that. If you have some documentation on how this is possible on the Solana network, I would be happy to review it.

Maybe it would make more sense for the user to sign a message and then your backend could receive that message and handle whatever transactions are necessary on behalf of the user?

**Comment by shirecoding** [comment]
Hi @adamdelphantom thanks for replying! 

Yep there was 2 ways to do it depending whether the backend does the partial sign and the user does the final sign or the backend does the partial sign and the user does the partial sign.

I choose to make the backend partial sign and the user in the client do the final sign as it was easier to implement (no need FE polling to see if the transaction has went through)

This is how i do it currently, it works including in tests, however the wallet shows that warning. 

https://solana.stackexchange.com/questions/9701/signtransaction-removes-partialsigned-signatures-making-it-impossible-to-sign-a

**Reply by adamdelphantom** [reply]
@shirecoding Do you have a repo or hosted dapp where we can reproduce this for investigation?
